### PR TITLE
Wong2 main: merged from original code

### PR DESCRIPTION
**Pull request by hunkim**
Could you please check? Now it supports chatgpt pro. 

Could you please check @joonhok-fittube

### CODE REVIEW

File: src/background/providers/chatgpt.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-async function request(token: string, method: string, path: string, data: unknown) {
+async function request(token: string, method: string, path: string, data?: unknown) {
   return fetch(`https://chat.openai.com/backend-api${path}`, {
     method,
     headers: {
       'Content-Type': 'application/json',
       Authorization: `Bearer ${token}`,
     },
-    body: JSON.stringify(data),
+    body: data === undefined ? undefined : JSON.stringify(data),
```
Comment (hunkim) [inline src/background/providers/chatgpt.ts:13]: This is a nice  check

**Comment by joonhok-fittube** [comment]
> Could you please check? Now it supports chatgpt pro.
> 
> Could you please check @joonhok-fittube

@hunkim Sure, I will have a look tonight or tomorrow morning.

**Comment by hunkim** [comment]
@joonhok-fittube check it out!

Let's roll the chatgpt pro version asap. It will take two days to get approval from google as well.


image

**Comment by joonhok-fittube** [comment]
@hunkim I checked the code but it's not clear how the frontend of ChatGPT Plus user communicates with the server and how the new change handles that. 

I need to sign up for ChatGPT Plus myself and see how it works. I contacted one of my friends in the U.S. and I think I can make the subscription soon. I will post my progress in a few hours.

**Review by joonhok-fittube — APPROVED** [review]
@hunkim 
Okay, I confirmed the correct model(model for paid users) is used when you are a Plus member. LTGM

**Comment by hunkim** [comment]
Can you merge?

**Pull request merged**